### ISSUE TITLE
Deprecate `master` branch completely (remove it) in 3 mounths

### ISSUE THREAD

**Issue by EmilySeville7cfg** [tldr-pages/tldr#9628]

**Comment by kbdharun** [comment]
I would suggest starting a discussion regarding this in our Gitter Chatroom.

**Comment by MasterOdin** [comment]
The transition to `main` happened around May 1, 2021 (https://github.com/tldr-pages/tldr/discussions/5868). I would propose that at this point `master` gets deprecated immediately to be removed on May 1, 2023, giving all clients a full 2 years since `main` was made available to make the transition. A follow-up would be to then go through the[ list of clients on the wiki](https://github.com/tldr-pages/tldr/wiki/tldr-pages-clients), and for any that are still using `master`, create an issue alerting them that `master` will be removed, and for them to transition.

**Comment by kbdharun** [comment]
Interestingly, Even our python client uses the `master` branch. I created this PR https://github.com/tldr-pages/tldr-python-client/pull/200 to update it.

**Comment by kbdharun** [comment]
I noticed some clients use the `master` version of the zip URL from https://github.com/tldr-pages/tldr-pages.github.io too. Should we keep the branch over there or depreciate it along with it?

**Comment by pixelcmtd** [comment]
> Interestingly, Even our python client uses the `master` branch.

I think I'll have to do another round of checking every `tldr` client before we can even think about removing `master`.

**Comment by kbdharun** [comment]
> > Interestingly, Even our python client uses the `master` branch.
> 
> I think I'll have to do another round of checking every `tldr` client before we can even think about removing `master`.

I am almost done, I will add the list here in an hour.

**Comment by kbdharun** [comment]
| **List of Clients referring to master branch or using it to fetch pages** |
|---------------------------------------------------------------------------|
| https://github.com/Moddus/tldr-python-dash-docset                         |
| https://github.com/principis/tldr-sharp                                   |
| https://github.com/ronan696/keypirinha-tldr                               |
| https://github.com/unInstance/untldr                                      |
| https://github.com/anoopengineer/tldr                                     |
| https://github.com/hterkelsen/tldr                                        |
| https://github.com/pepa65/tldr-bash-client                                |
| https://github.com/avih/tldr-fmt                                          |
| https://github.com/BrainMaestro/tldr-php                                  |
| https://github.com/kirillseva/tldrrr                                      |
| https://github.com/k3mist/tldr                                            |
| https://github.com/tldr-pages/extldr                                      |
| https://github.com/mflint/ios-tldr-viewer                                 |
| https://github.com/YellowApple/tldrb                                      |
| https://gitlab.com/ve-nt/outfieldr                                        |
| https://github.com/iamgodot/py-tldr                                       |
| https://github.com/HardDie/myTldr                                         |
| https://github.com/luthfianto/rust-tldr                                   |
| https://github.com/wlemuel/vim-tldr                                       |
| https://github.com/raylee/tldr-sh-client                                  |

**Comment by kbdharun** [comment]
Done you can check the above list @pixelcmtd

**Comment by pixelcmtd** [comment]
> Done you can check the above list

@kbdharun Just saw this now, gg I've wasted the last 2 or so hours of my life :c

**Comment by kbdharun** [comment]
> > Done you can check the above list
> 
> @kbdharun Just saw this now, gg I've wasted the last 2 or so hours of my life :c

Oh, 😓 🤣

**Comment by unInstance** [comment]
Thank you! I have merged your pull request.

**Comment by kbdharun** [comment]
Reminded, the repo owners of the unmerged PRs about the change, I will do it again after 15 days.

**Comment by pixelcmtd** [comment]
It's 1 May 04:25 in UTC right now. Does anyone have further concerns or can we delete the `master` branch in a few hours?

**Comment by kbdharun** [comment]
> It's 1 May 04:25 in UTC right now. Does anyone have further concerns or can we delete the `master` branch in a few hours?

Can we start a discussion regarding this in the chatroom, in order to get others'opinions too.

**Comment by pixelcmtd** [comment]
> I think we could do it, and update wiki to reflect the remaining clients here as borked or outdated.

I've already spent my last few days thinking about whether to make a table of unmaintained, broken and outdated clients or just mark and move them to the ends of their respective tables.

Could also tie in well with the client spec compliance/certification plans I've been thinking about proposing for months (maybe even years, I've lost track) by now

**Comment by kbdharun** [comment]
> > I think we could do it, and update wiki to reflect the remaining clients here as borked or outdated.
> 
> I've already spent my last few days thinking about whether to make a table of unmaintained, broken and outdated clients or just mark and move them to the ends of their respective tables.

Yeah that would be the best

> 
> Could also tie in well with the client spec compliance/certification plans I've been thinking about proposing for months (maybe even years, I've lost track) by now

Agreed, we should mark if the community clients conform to the client specification too. 

@SethFalco and @waldyrious what do you think?

**Comment by pixelcmtd** [comment]
Oh god, do you really want to turn this into a discussion about that? I was very not ready for that yet, but whatever. So my ideas were basically something along these lines:

- Figure out whether we can put the client spec into a machine-readable/algorithmic/? form
- If yes, do that. Otherwise, make a checklist at the end of the spec with a lot of commands and what they're supposed to do
- Optionally, make a tool that you can give the path to a tldr client and it tells you whether it's compliant (this could probably be a 10 line shell script)
- Sort the client list by compliance
- Spend like 10 minutes of design work making certificate "badges"
- Allow clients that are 100% compliant with a version of the client spec to carry the respective badge (after they've been certified by someone from tldr; each client vendor can open an issue if they think they're compliant and if 2 tldr members/contributors sign off on it, they are listed on the wiki with that compliance and afterwards allowed to carry that badge when advertising their project etc)

**Comment by kbdharun** [comment]
> Allow clients that are 100% compliant with a version of the client spec to carry the respective badge (after they've been certified by someone from tldr; each client vendor can open an issue if they think they're compliant and if 2 tldr members/contributors sign off on it, they are listed on the wiki with that compliance and afterwards allowed to carry that badge when advertising their project etc)

I have been thinking for a while, and it would make sense with this change. How about we limit Wiki editing to collaborators and add an issue template for client authors to request their Client to be added/update details in Wiki?

**Comment by pixelcmtd** [comment]
> How about we limit Wiki editing to collaborators

Why the f- isn't it right now? Seems kinda obvious, sure.

> How about we [...] add an issue template for client authors to request their Client addition or if it's already present then updation of details in the Wiki?

Taking a more incremental approach, we should probably make such an issue template right now.

**Comment by SethFalco** [comment]
> table of unmaintained, broken and outdated clients or just mark and move them to the ends of their respective tables.

Let's not move them to the end of a table.
They should be separated entirely, and out of sight for new users just looking for a client. It'd be better, imo to have the list of maintained clients that follow our client specifications, and then a separate table (in a `details` tag) of unmaintained clients.

> Figure out whether we can put the client spec into a machine-readable/algorithmic/? form

We can use [Bats](https://github.com/bats-core/bats-core). It's a test framework that enables us to define test suites for bash. In doing so, we can create tests that verify clients meet the interface required by our client specifications. We can test against supported arguments, commands, or even the layout of the output.

> Why the f- isn't it right now? Seems kinda obvious, sure.

Because by definition, a "wiki" can be edited by the community.

> A collaborative website whose content can be edited by anyone who has access to it.
> 
> — https://www.wordnik.com/words/wiki

If something were to be obvious, it would be that it should be open to the public. Though I have nothing against limiting who has write-access to the wiki, I just want to note why it's not "obvious".

**Comment by SethFalco** [comment]
Just a note on the side.

> Oh god, do you really want to turn this into a discussion about that?

> Why the f- isn't it right now?

@pixelcmtd, watch your attitude.

**Comment by waldyrious** [comment]
I agree that the default-to-open aspect of wikis would be the natural expectation. If GitHub's implementation of the concept makes it hard to detect spam edits (which seems to be the case), and if such edits are prevalent (are they?), then I would agree, albeit reluctantly, that it makes sense to close it down a bit.

As for splitting the unmaintained clients — on the one hand, it pains me to contribute to the march towards obsolescence of software that was perfectly functional in the past, and had the ecosystem around it change from underneath their feet, but I suppose that process is both inevitable and to some degree desirable, lest we block progress in the name of conservatism. Blabbering aside, tl;dr: yeah, I won't object to making the move of abandoning the master branch and de-recognizing clients that haven't updated to support the new branch name.

I would prefer if a broader discussion about compliance to the spec was handled in a separate thread, so that this one could be closed once the two steps I mentioned above are taken. Otherwise I'm afraid this already long thread will drag on for even longer.

**Comment by pixelcmtd** [comment]
Given our usual 12 hour waiting period, I'd suggest removing the branch at 17:00 (or 5 PM if you're wrong) UTC, or 2 hours from now

**Comment by pixelcmtd** [comment]
<img width="278" alt="`master` Deleted just now by pixelcmtd" src="https://user-images.githubusercontent.com/35269695/235496775-c066dad0-2d86-4c37-8eb4-7f1bdac3f2d3.png">

I've moved the broken clients from [the comment above](https://github.com/tldr-pages/tldr/issues/9628#issuecomment-1365912040) [to the bottom](https://github.com/tldr-pages/tldr/wiki/tldr-pages-clients/e62c2de9f7c0e02c2b52940daab201a20512fcef). There are probably more, especially ones that have been archived for a while.
